pkg/tests/tasks/security/authorization/http_test.go

```diff
@@ -16,105 +16,230 @@ package authorization
-		body, err := ioutil.ReadAll(resp.Body)
-		util.Inspect(err, "Failed to read response body", "", t)
-		if strings.Contains(string(body), "RBAC: access denied") {
-			log.Log.Infof("Got access denied as expected: %s", string(body))
-		} else {
-			t.Errorf("RBAC deny all failed. Got response: %s", string(body))
-			log.Log.Errorf("RBAC deny all failed. Got response: %s", string(body))
-		}
-		util.CloseResponseBody(resp)
-	})
-
-	t.Run("Security_authorization_rbac_allow_GET_http", func(t *testing.T) {
-		defer util.RecoverPanic(t)
-
-		log.Log.Info("Allow access with GET method to the productpage workload")
-		util.KubeApplyContents("bookinfo", ProductpageGETPolicy)
-		time.Sleep(time.Duration(10) * time.Second)
-
-		util.GetHTTPResponse(productpageURL, nil) // dummy request to refresh previous page
-		resp, _, err := util.GetHTTPResponse(productpageURL, nil)
-		util.Inspect(err, "Failed to get HTTP Response", "", t)
-		body, err := ioutil.ReadAll(resp.Body)
-		util.Inspect(err, "Failed to read response body", "", t)
-		if strings.Contains(string(body), "Error fetching product details") && strings.Contains(string(body), "Error fetching product reviews") {
-			log.Log.Infof("Got expected page with Error fetching product details and Error fetching product reviews")
-		} else {
-			t.Errorf("Productpage GET policy failed. Got response: %s", string(body))
-			log.Log.Errorf("Productpage GET policy failed. Got response: %s", string(body))
-		}
-		util.CloseResponseBody(resp)
-
-		log.Log.Info("Allow other bookinfo services GET method")
-		util.KubeApplyContents("bookinfo", DetailsGETPolicy)
-		util.KubeApplyContents("bookinfo", ReviewsGETPolicy)
-		util.KubeApplyContents("bookinfo", RatingsGETPolicy)
-		time.Sleep(time.Duration(50) * time.Second)
-
-		util.GetHTTPResponse(productpageURL, nil) // dummy request to refresh previous page
-		resp, _, err = util.GetHTTPResponse(productpageURL, nil)
-		util.Inspect(err, "Failed to get HTTP Response", "", t)
-
-		body, err = ioutil.ReadAll(resp.Body)
-		util.Inspect(err, "Failed to read response body", "", t)
-		if strings.Contains(string(body), "Error fetching product details") || strings.Contains(string(body), "Error fetching product reviews") || strings.Contains(string(body), "Ratings service currently unavailable") {
-			t.Errorf("GET policy failed. Got response: %s", string(body))
-			log.Log.Errorf("GET policy failed. Got response: %s", string(body))
-		} else {
-			log.Log.Infof("Got expected page.")
-		}
-		util.CloseResponseBody(resp)
+func TestAuthoriztionHTTPTraffic(t *testing.T) {
+	test.NewTest(t).Id("T20").Groups(test.Full, test.ARM, test.InterOp).Run(func(t test.TestHelper) {
+		hack.DisableLogrusForThisTest(t)
+
+		ns := "bookinfo"
+		t.Cleanup(func() {
+			oc.RecreateNamespace(t, ns)
+			shell.Execute(t,
+				fmt.Sprintf(`kubectl patch -n %s smcp/%s --type merge -p '{"spec":{"security":{"dataPlane":{"mtls":false},"controlPlane":{"mtls":false}}}}'`,
+					meshNamespace, smcpName))
```

Comment:
Can replace this with `shell.Executef` to shorten it a bit, or even better, use `oc.MergePatch` (which Francisco forgot to rename to just `Patch`).